staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1212,10 +1190,30 @@ func (e *Store) DeleteCollection(ctx context.Context, deleteValidation rest.Vali
 			}
 		}()
 	}
-	wg.Wait()
-	// notify distributor to exit
-	close(workersExited)
-	<-distributorExited
+	// In case of all workers exit, notify distributor.
+	go func() {
+		defer utilruntime.HandleCrash(func(panicReason interface{}) {
+			errs <- fmt.Errorf("DeleteCollection distributor panicked: %v", panicReason)
```

Comment:
Is this actually the distributor that panicked?

Reply:
fair - fixed